Reuse dot_clk input as externel reset input

niclashoyer opened the pull request:
This changes reuse the dot_clk input on rev 0.2x PCBs to sense the /RESET line of the C64. With this the TeensyROM gets back into the menu when an internal reset is fired in the C64 (e.g. another reset button or other devices like blue64). The trace from dot_clk to U4 needs to be cut and a jumper wire from RESET to the right pin on U4.

To use this enable `DbgSignalSenseReset` in `TeensyROM.h` and `Min_TeensyROM.h`.

niclashoyer committed:
Reuse dot_clk input as externel reset input

SensoriumEmbedded commented:
Thank you for this submission, apologies for the delayed response.
I don't think I can incorporate this change as written, since it requires HW modification.  It also won't work with rev 0.3 of the PCA since that signal is re-used.
I would like to leave this here/available though as I do like the idea and hope to incorporate this into a future revision of the PCB.
Thank you for your support!  :)

niclashoyer commented:
@SensoriumEmbedded yes, sure. I just got this to build and ported your batch files to the following script that roughly does the same (just using `cp`, `head` and `tail`):

<details>

<summary>build-helper.sh</summary>

```bash
#!/bin/bash

COREPATH="$HOME/.arduino15/packages/teensy/hardware/avr/1.59.0/cores/teensy4"

set -e

function upper() {
  cp ./BootLinkerFiles/bootdata.c.upper $COREPATH/bootdata.c
  cp ./BootLinkerFiles/imxrt1062_t41.ld.upper $COREPATH/imxrt1062_t41.ld
  echo "Moved address to upper position. Now please compile Teensy.ino"
}

function orig() {
  cp ./BootLinkerFiles/bootdata.c.orig $COREPATH/bootdata.c
  cp ./BootLinkerFiles/imxrt1062_t41.ld.orig $COREPATH/imxrt1062_t41.ld
  echo "Moved address to original position. Now please compile MinimalBoot.ino"
}

function merge() {
  head -n -2 MinimalBoot.ino.hex > merged.hex
  head -n -2 Teensy.ino.hex >> merged.hex
  tail -2 MinimalBoot.ino.hex >> merged.hex
  echo "Merged upper/lower parts into merged.hex"
}

case "$1" in
  upper)
    upper
    ;;

  orig)
    orig
    ;;

  merge)
    merge
    ;;

  ?|help)
    echo "Usage: $(basename $0) upper|orig|merge"
    exit 1
    ;;
esac
```

</details>

```bash
$ ./build-helper.sh upper
Moved address to upper position. Now please compile Teensy.ino
$ ./build-helper.sh orig
Moved address to original position. Now please compile MinimalBoot.ino
# copy Teensy.ino.hex and MinimalBoot.ino.hex to this directory
$ ./build-helper.sh merge
Merged upper/lower parts into merged.hex
```

niclashoyer commented:
I just updated the PR to include the same reset changes to minimal boot, so after loading a large CRT the external reset will also boot the menu again.

[TeensyROM.0.7_full+2_RESET_mod.hex.zip](https://github.com/user-attachments/files/24357279/TeensyROM.0.7_full%2B2_RESET_mod.hex.zip)

niclashoyer commented:
Latest build with fixes applied: 
[TeensyROM.0.7_full+2_RESET_mod.hex.zip](https://github.com/user-attachments/files/24361829/TeensyROM.0.7_full%2B2_RESET_mod.hex.zip)

SensoriumEmbedded commented:
Looks great, merging now.  Thank you very much @niclashoyer !
